src/openvic-simulation/pop/Pop.cpp

```diff
@@ -132,6 +276,29 @@ bool PopManager::add_pop_type(
 void PopManager::reserve_pop_types(size_t count) {
 	stratas.reserve(stratas.size() + count);
 	pop_types.reserve(pop_types.size() + count);
+	delayed_parse_nodes.reserve(delayed_parse_nodes.size() + count);
+}
+
+static NodeCallback auto expect_needs_income(PopType::income_type_t& types) {
+	static const string_map_t<PopType::income_type_t> income_type_map {
+		{ "administration", ADMINISTRATION },
+		{ "education", EDUCATION },
+		{ "military", MILITARY },
+		{ "reforms", REFORMS }
+	};
+	return expect_dictionary_keys(
+		"type", ONE_OR_MORE, expect_identifier(expect_mapped_string(income_type_map,
+			[&types](PopType::income_type_t type) -> bool {
+				if (!share_income_type(types, type)) {
+					types |= type;
+					return true;
+				}
+				Logger::error("Duplicate income type ", type, " in pop type income types!");
+				return false;
+			}
+		)),
+		"weight", ZERO_OR_ONE, success_callback /* Has no effect in game */
```

Comment:
Should we add a warning for weight != 1?
If it is added with a different value then the modder could be informed that it has no effect.

Reply:
Vanilla adds it with values other than 1 (e.g. soldiers have 1.5), it's just gonna create warning spam.